### DISCUSSION TITLE
Server side page components not refreshing when Link is clicked

### DISCUSSION THREAD

**Discussion by jfoxworth** [vercel/next.js#48312] [Help]
### Summary

I have a page that renders server side. On that page is a series of links which will take the user to the same page but with new URL parameters. Think of it as a folder page that shows child folders. When the user clicks on one of those folders, it should populate the page's components with the new data.

The problem that I am having is that when a Link is clicked, the URL updates and getServerSideProps gets the new data. However, the page does not re render.

You can see the code on stack overflow.

https://stackoverflow.com/questions/75991442/next-js-clicking-link-on-page-prompts-getserversideprops-but-does-not-refresh-c

I feel like I am missing something. If the intent is for SSR pages to always behave this way, then they seem to negate the core use of react - updating components as opposed to reloading the whole page?

### Example

https://stackoverflow.com/questions/75991442/next-js-clicking-link-on-page-prompts-getserversideprops-but-does-not-refresh-c

**Comment by HastaCs** [comment]
i was the same problem, and i fixed puting this is my mian layout.tsx

import { revalidatePath } from 'next/cache';
revalidatePath('/', 'layout')

https://nextjs.org/docs/app/api-reference/functions/revalidatePath
